frame/treasury/src/lib.rs

```diff
@@ -233,6 +132,17 @@ pub trait Trait<I=DefaultInstance>: frame_system::Trait {
 
 	/// Weight information for extrinsics in this pallet.
 	type WeightInfo: WeightInfo;
+
+	/// Runtime hooks to external pallet using treasury to compute spend funds.
+	type SpendFunds: SpendFunds<Self, I>;
+}
+
+#[impl_trait_for_tuples::impl_for_tuples(30)]
+pub trait SpendFunds<T: Trait<I>, I=DefaultInstance> {
+    fn spend_funds( budget_remaining: &mut BalanceOf<T, I>,
+                    imbalance: &mut PositiveImbalanceOf<T, I>,
+                    total_weight: &mut Weight,
+                    missed_any: &mut bool );
 }
```

Comment:
For experts in this macro, how can we make this function signature better? This seems like not the right final approach...